docker/Dockerfile

```
@@ -1,36 +1,56 @@
+    git fetch -f --tags github-upstream
 # Ignore error here. This always fails on Docker Cloud. It's fine there because the submodule is alread initialized. See also:
 # https://stackoverflow.com/questions/58690455/how-to-correctly-initialize-git-submodules-in-dockerfile-for-docker-cloud
 RUN git submodule update --init --recursive || true
 
 WORKDIR /opt/open62541/build
 RUN cmake -DBUILD_SHARED_LIBS=ON \
-		-DCMAKE_BUILD_TYPE=Debug \
-		-DUA_BUILD_EXAMPLES=ON \
-		# Hardening needs to be disabled, otherwise the docker build takes too long and travis fails
-		-DUA_ENABLE_HARDENING=OFF \
-        -DUA_ENABLE_ENCRYPTION=MBEDTLS \
-        -DUA_ENABLE_SUBSCRIPTIONS=ON \
-        -DUA_ENABLE_SUBSCRIPTIONS_EVENTS=ON \
-		-DUA_NAMESPACE_ZERO=FULL \
-         /opt/open62541
-RUN make -j
-RUN make install
+      -DCMAKE_BUILD_TYPE=Release \
+      -DUA_BUILD_EXAMPLES=ON \
+      # Hardening needs to be disabled, otherwise the docker build takes too long and travis fails
+      -DUA_ENABLE_HARDENING=OFF \
+      -DUA_ENABLE_ENCRYPTION=MBEDTLS \
+      -DUA_ENABLE_SUBSCRIPTIONS=ON \
+      -DUA_ENABLE_SUBSCRIPTIONS_EVENTS=ON \
+      -DUA_NAMESPACE_ZERO=FULL \
+      /opt/open62541 && \
+    make -j && \
+    make install
 WORKDIR /opt/open62541
 
 # Generate certificates
-RUN apk add --no-cache python3-dev linux-headers openssl && rm -rf /var/cache/apk/*
-RUN pip3 install --no-cache-dir netifaces==0.10.9
-RUN mkdir -p /opt/open62541/pki/created
-RUN python3 /opt/open62541/tools/certs/create_self-signed.py /opt/open62541/pki/created
+RUN apk add --no-cache python3-dev linux-headers openssl && rm -rf /var/cache/apk/* && \
+    pip3 install --no-cache-dir netifaces==0.10.9 && \
+    mkdir -p /opt/open62541/pki/created && \
+    python3 /opt/open62541/tools/certs/create_self-signed.py /opt/open62541/pki/created
 
 
+FROM alpine:3.13.6 AS server
+
+RUN apk add --no-cache mbedtls-dev && \
+    mkdir -p /opt/open62541/pki/created
+
+COPY --from=build /opt/open62541/pki/created /opt/open62541/pki/created
+COPY --from=build /opt/open62541/build/bin/examples /opt/open62541/examples
+COPY --from=build /usr/local/lib64/libopen62541* /usr/local/lib64/
+COPY --from=build /usr/local/lib64/cmake/open62541/ /usr/local/lib64/cmake/open62541/
+COPY --from=build /usr/local/lib64/pkgconfig/open62541.pc /usr/local/lib64/pkgconfig/open62541.pc
+COPY --from=build /usr/local/share/open62541/tools/certs /usr/local/share/open62541/tools/certs
+COPY --from=build /usr/local/include/open62541/ /usr/local/include/open62541/
+COPY --from=build /usr/local/include/ms_stdint.h /usr/local/include/ms_stdint.h
+COPY --from=build /usr/local/include/ziptree.h /usr/local/include/ziptree.h
+COPY --from=build /usr/local/include/aa_tree.h /usr/local/include/aa_tree.h
+COPY --from=build /usr/local/bin/ua_server_ctt.exe /usr/local/bin/ua_server_ctt.exe
+COPY --from=build /usr/local/bin/ua_client /usr/local/bin/ua_client
```

We can get rid of all the explicit COPY layers and just copy over all necessary collected files in a single mount-and-copy layer.

```suggestion
RUN --mount=src=/release,dst=/release,from=collect \
    cp -r /release/* /
```

---

Changed in 946115aab12994a36174c0b0877ca7cc73cf0768